src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -2639,22 +2639,27 @@ private void normalizeMaximumResponseEntities(
     private void makeEmailStateConsistent(FeedbackSessionAttributes oldSession,
             FeedbackSessionAttributes newSession) {
 
-        // reset sentOpenEmail if the session has opened but is being closed
-        // now.
-        if (oldSession.isSentOpenEmail() && !newSession.isOpened()) {
-            newSession.setSentOpenEmail(false);
-        } else if (oldSession.isSentOpenEmail()) {
-            // or else leave it as sent if so.
-            newSession.setSentOpenEmail(true);
+        // reset sentOpenEmail if the session has opened but is being un-opened
+        // now, or else leave it as sent if so.
+        if (oldSession.isSentOpenEmail()) {
+            newSession.setSentOpenEmail(newSession.isOpened());
+        }
+
+        // reset sentClosedEmail if the session has closed but is being un-closed
+        // now, or else leave it as sent if so.
+        if (oldSession.isSentClosedEmail()) {
+            newSession.setSentClosedEmail(newSession.isClosed());
+            
+            // also reset sentClosingEmail
+            newSession.setSentClosingEmail(
+                    newSession.isClosed()
+                    || !newSession.isClosingWithinTimeLimit(SystemParams.NUMBER_OF_HOURS_BEFORE_CLOSING_ALERT));
```

Comment:
The code for `isClosingWithinTimeLimit()` appears to check if the session is closing between 23 to 24 hours (given the current parameters) from now, and may result in incorrectly setting this value.

Example cases:
Session has closed - `sentClosingEmail` is true (expected)
Session is closing in 10 hours - `sentClosingEmail` is true (expected)
Session is closing in 23.5 hours - `sentClosingEmail` is false (expected? feels like it's the correct output, but I'm not certain)
Session is closing in 48 hours - `sentClosingEmail` is true (unexpected)